lib/code.rb

```diff
@@ -0,0 +1,24 @@
+class Code
+  def self.winner?(arr, player)
+    return true if [
+      [arr[0], arr[1], arr[2]].all?(player),
+      [arr[3], arr[4], arr[5]].all?(player),
+      [arr[6], arr[7], arr[8]].all?(player),
+
+      [arr[0], arr[3], arr[6]].all?(player),
+      [arr[1], arr[4], arr[7]].all?(player),
+      [arr[2], arr[5], arr[8]].all?(player),
+
+      [arr[0], arr[4], arr[8]].all?(player),
+      [arr[6], arr[4], arr[2]].all?(player)
+    ].any?
+
+    false
+  end
```

Comment:
It can be simplified. For example
```ruby
def check_win(player_symbol)
    # for rows
    all_equal(board[0..2], player_symbol) ||
    all_equal(board[3..5], player_symbol) ||
    all_equal(board[6..8], player_symbol)

    # and analogical as above for columns and diagonals
  end

  def all_equal(subarray, player_symbol)
    return true if subarray.uniq.size == 1 && subarray.uniq == player_symbol
  end
```